modules/gateway/nodes.go

```diff
@@ -16,18 +16,29 @@ var (
 	errOurAddress = errors.New("can't add our own address")
 )
 
+type node struct {
+	Inbound bool `json:"inbound"`
```

Comment:
A json tag makes sense if this object is being saved to disk, but it looks like you have the `nodePersist` type for that.

Given that `addNode` is now a bit ugly (taking two args), I propose that we define:
```go
type node struct { 
    Addr     modules.NetAddress `json:"addr"`
    Outbound bool        `json:"outbound"`
}
```
And have `g.addNode` take a `node` (maybe a `*node`, not sure) instead of two args.
Lastly, I think it's fine to save this object in the gateway persist directly (instead of having a separate `nodePersist` type). The `g.nodes` map should be flattened to a `[]node` during the save.